automerge-backend/src/object_store.rs

```diff
@@ -52,6 +60,50 @@ impl ListState {
         }
     }
 
+    fn generate_diffs(&self) -> Vec<Diff> {
+        let mut diffs = Vec::new();
+
+        let head = Diff {
+            action: DiffAction::CreateList(self.object_id.clone(), self.sequence_type.clone()),
+            conflicts: Vec::new(),
+        };
+
+        let ops_in_order = list_ops_in_order(&self.operations_by_elemid, &self.following)
+            .ok()
+            .unwrap_or_default();
+
+        let inserts = ops_in_order
+            .iter()
+            .rev()
+            .enumerate()
```

Comment:
Does this `enumerate` call need to be after we filter out the deleted elements (via `filter_map` on `op.active_op()`, which will be `None` if there are no operations a la delete)? I think we might end up with incorrect element indices in the case of deleted elements?

Reply:
Should be fixed now.  Had to add an unwrap()